src/FSharp.Control.Websockets.TPL/FSharp.Control.Websockets.TPL.fs

```diff
@@ -118,16 +119,25 @@ module ThreadSafeWebsocket =
                 let! message = sendBuffer.ReceiveAsync()
                 match message with
                 | Send (cancellationToken, buffer, messageType, stream, replyChannel) ->
-                    do! Websocket.sendMessage cancellationToken buffer messageType stream webSocket
-                    replyChannel.SetResult ()
+                    try
+                        do! Websocket.sendMessage cancellationToken buffer messageType stream webSocket
+                        replyChannel.SetResult (Ok ())
+                    with
+                    | ex -> replyChannel.SetResult (Error ex)
                 | Close (cancellationToken, status, message, replyChannel) ->
-                    hasClosedBeenSent <- true
-                    do! webSocket.CloseAsync(status,message,cancellationToken)
-                    replyChannel.SetResult ()
+                    try
+                        hasClosedBeenSent <- true
+                        do! webSocket.CloseAsync(status,message,cancellationToken)
+                        replyChannel.SetResult (Ok ())
+                    with
+                    | ex -> replyChannel.SetResult (Error ex)
```

Comment:
Should we use the `TaskCompletionSource`'s set exception instead?

Reply:
I don't think so, because that leads to a caller having to try/catch. We want to represent the failure in the type system, I think, so that the caller has to handle the error case. Also as mentioned on slack I need to more properly capture the stack trace using the DispatchInfo.

Reply:
👍